core/workspace.js

```diff
@@ -193,6 +204,61 @@ Blockly.Workspace.prototype.getTopBlocks = function(ordered) {
   return blocks;
 };
 
+/**
+ * Add a comment to the list of top comments.
+ * @param {!Blockly.WorkspaceComment} comment comment to add.
+ * @package
+ */
+Blockly.Workspace.prototype.addTopComment = function(comment) {
+  this.topComments_.push(comment);
+
+  // Note: If the comment database starts to hold block comments, this may need
+  // to move to a separate function.
+  if (this.commentDB_[comment.id]) {
+    console.warn('Overriding an existing comment on this workspace, with id "' +
+        comment.id + '"');
```

Comment:
Does this basically happen every time we change the (text) contents of an existing comment? Or is this covering some case I'm missing?

Reply:
Mostly I meant to ask what's the use case for comment IDs colliding?